ouroboros-consensus/src/Data/SOP/Strict.hs

```diff
@@ -60,6 +62,19 @@ type instance AllZipN    NP c = AllZip c
 type instance Prod       NP   = NP
 type instance SListIN    NP   = SListI
 
+instance forall k xs . (Typeable (NP k xs), All (Compose ToCBOR k) xs) => ToCBOR (NP k xs) where
```

Comment:
I have never seen a `forall` like this on an instance until last week. And now here it is again!

I do wonder: maybe we don't need it? `instance` already binds the free type variables. (I haven't double-checked it but I'm almost certain; else I would have known you could put `forall` here because I would have needed it.)

Not a major comment here.